src/components/List/List.tsx

```diff
@@ -131,20 +141,25 @@ class List extends UIComponent<Extendable<ListProps>, ListState> {
 
   renderItems() {
     const { items } = this.props
-    const { selectedItemIndex } = this.state
+    const { focusedIndex, selectedIndex } = this.state
 
     this.itemRefs = []
 
     return _.map(items, (item, idx) => {
       const maybeSelectableItemProps = {} as any
 
-      if (this.props.selection) {
+      if (this.props.selectable) {
         const ref = React.createRef()
         this.itemRefs[idx] = ref
 
-        maybeSelectableItemProps.tabIndex = idx === selectedItemIndex ? 0 : -1
         maybeSelectableItemProps.ref = ref
-        maybeSelectableItemProps.onFocus = () => this.focusHandler.syncFocusedItemIndex(idx)
+        maybeSelectableItemProps.onFocus = () => this.focusHandler.syncFocusedIndex(idx)
+        maybeSelectableItemProps.onClick = () => {
+          this.trySetState({ selectedIndex: idx })
+          this.setState({ focusedIndex: idx })
```

Comment:
note that at this moment you cannot be sure whether `selectedIndex` had been set to `idx` or not. Consider controlled mode example: when `trySetState` is called, there should be corresponding event fired (`onSelectedIndexChange` - in fact, this is missed from the API at the moment) that will notify client that component would like to change this prop - but it will be up to the client to decide whether its value should be updated, where client may just decide to deny update.

Reply:
Added ```onSelectedIndexChange```